py/h2o_wave/h2o_wave/cli.py

```diff
@@ -143,40 +144,50 @@ def run(app: str, no_reload: bool, no_autostart: bool):
 
     # Try to start Wave daemon if not running or turned off.
     server_port = int(os.environ.get('H2O_WAVE_LISTEN', ':10101').split(':')[-1])
-    server_not_running = _scan_free_port(server_port) == server_port
+    server_running = _scan_free_port(server_port) != server_port
 
     waved_process = None
     if no_autostart:
         autostart = False
     else:
         autostart = os.environ.get('H2O_WAVE_NO_AUTOSTART', 'false').lower() in ['false', '0', 'f']
 
-    waved_path = os.path.join(sys.exec_prefix, 'waved.exe' if IS_WINDOWS else 'waved')
+    waved_cwd = sys.exec_prefix
+    proc_name = 'waved.exe' if IS_WINDOWS else 'waved'
+    # If waved is running, find its cwd.
+    if server_running:
+        for proc in psutil.process_iter():
+            if proc.name() == proc_name:
+                try:
+                    waved_cwd = psutil.Process(proc.pid).cwd()
+                except psutil.NoSuchProcess:
+                    pass
+    waved_path = os.path.join(waved_cwd, proc_name)
```

Comment:
This is only true for platform specific wheel files. Platform-agnostic whl does not contain waved binary.